README.md

```diff
@@ -3,48 +3,124 @@ Manages the lifecycle of a NATS JetStream deployment.
 
 ## Description
 It is a standard Kubernetes operator which observes the state of NATS JetStream deployment and reconciles its state according to desired state.
-## Getting Started
-You’ll need a Kubernetes cluster to run against. You can use [k3d](https://k3d.io/) to get a local cluster for testing, or run against a remote cluster.
-**Note:** Your controller will automatically use the current context in your kubeconfig file (i.e. whatever cluster `kubectl cluster-info` shows).
 
-### Running on the cluster
-1. Install Instances of Custom Resources:
+### How it works
+
+This project aims to follow the Kubernetes [Operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/).
+
+It uses [Controllers](https://kubernetes.io/docs/concepts/architecture/controller/),
+which provide a reconcile function responsible for synchronizing resources until the desired state is reached on the cluster.
+
+## Development
+
+### Pre-requisites
 
+- [Go](https://go.dev/)
+- [Docker](https://www.docker.com/)
+- [kubectl](https://kubernetes.io/docs/tasks/tools/)
+- [kubebuilder](https://book.kubebuilder.io/)
+- [kustomize](https://kustomize.io/)
+- Access to Kubernetes cluster ([k3d](https://k3d.io/) / k8s)
+
+### Running locally
+1. Install the CRDs into the cluster:
+    ```sh
+    make install
+    ```
+
+2. Run your controller (this will run in the foreground, so switch to a new terminal if you want to leave it running):
+    ```sh
+    make run
+    ```
+
+    **NOTE:** You can also run this in one step by running: `make install run`
+
+### Running tests
+Run the unit and integration tests:
 ```sh
-kubectl apply -f config/samples/
+make test-only
 ```
 
-2. Build and push your image to the location specified by `IMG`:
+### Linting
+1. Fix common lint issues:
+    ```sh
+    make imports-local
+    make fmt-local
+    ```
+
+2. Run lint check:
+    ```sh
+    make lint-thoroughly
+    ```
+
+### Modifying the API definitions
+If you are editing the API definitions, generate the manifests such as CRs or CRDs using:
+
+```sh
+make manifests
+```
+
+**NOTE:** Run `make --help` for more information on all potential `make` targets
+
+More information can be found via the [Kubebuilder Documentation](https://book.kubebuilder.io/introduction.html)
+
+### Build container images
+
+Build and push your image to the location specified by `IMG`:
 
 ```sh
 make docker-build docker-push IMG=<some-registry>/nats-manager:tag
```

Comment:
`tag` is supposed to be the (version) tag you give the image, right? If you're supposed to put a value there yourself, I think markers like `<tag>` would be nice

Reply:
updated